cluster-autoscaler/core/scale_down.go

```diff
@@ -572,6 +572,46 @@ func (sd *ScaleDown) mapNodesToStatusScaleDownNodes(nodes []*apiv1.Node, nodeGro
 	return result
 }
 
+// SoftTaintUnneededNodes manage soft taints of unneeded nodes.
+func (sd *ScaleDown) SoftTaintUnneededNodes(allNodes []*apiv1.Node) (errors []error) {
+	apiCallBudget := sd.context.AutoscalingOptions.MaxBulkSoftTaint
+	// Check if some new nodes should be tainted
+	for _, node := range allNodes {
+		if deletetaint.HasToBeDeletedTaint(node) {
+			// Do not consider nodes that are scheduled to be deleted
+			continue
+		}
+		alreadyTainted := deletetaint.HasDeletionCandidateTaint(node)
+		_, unneeded := sd.unneededNodes[node.Name]
+
+		if unneeded != alreadyTainted {
```

Comment:
Maybe it is just me but I would prefer to have this if split into two:

```
if unneeded && !alreadyTainted  ...
if !unneeded && alreadyTainted ...
```

Reply:
But there is a common block at the top of that if ...

Reply:
Yeah - still I would rather have it repeated.

Reply:
done